cloud-sql/mysql/Startup.cs

```diff
@@ -55,34 +54,100 @@ public void ConfigureServices(IServiceCollection services)
                     options.Version = "Test";
                 });
             }
-            services.AddScoped(typeof(DbConnection), (IServiceProvider) =>
-                InitializeDatabase());
+            services.AddScoped(typeof(MySqlConnectionStringBuilder),
+                (IServiceProvider) => GetMySqlConnectionString());
+            services.AddScoped<MySqlConnect>();
             services.AddMvc(options =>
             {
                 options.Filters.Add(typeof(DbExceptionFilterAttribute));
             });
             _services = services;
         }
 
-        DbConnection InitializeDatabase()
+        public class MySqlConnect
         {
-            DbConnection connection;
+            public DbConnection GetMySqlConnection(
+                MySqlConnectionStringBuilder connectionString)
+            {
+                // [START cloud_sql_mysql_dotnet_ado_backoff]
+                var connection = Policy
+                    .HandleResult<DbConnection>(conn => conn.State != ConnectionState.Open)
+                    .WaitAndRetry(new[]
+                    {
+                    TimeSpan.FromSeconds(1),
+                    TimeSpan.FromSeconds(2),
+                    TimeSpan.FromSeconds(5)
+                    }, (result, timeSpan, retryCount, context) =>
+                    {
+                        // Log any warnings here.
+                    })
+                    .Execute(() =>
+                    {
+                        // Return a new connection.
+                        // [START_EXCLUDE]
+                        if (Environment.GetEnvironmentVariable("DB_HOST") != null)
+                        {
+                            return NewMysqlTCPConnection(connectionString);
+                        }
+                        else
+                        {
+                            return NewMysqlUnixSocketConnection(connectionString);
+                        }
+                        // [END_EXCLUDE]
+                    });
+                // [END cloud_sql_mysql_dotnet_ado_backoff]
+                return connection;
+            }
+
+            DbConnection NewMysqlTCPConnection(
+                MySqlConnectionStringBuilder connectionString)
+            {
+                // [START cloud_sql_mysql_dotnet_ado_connection_tcp]

```

Comment:
In opinion the really important thing to show in these methods is how to build the connection string, not how to build a connection from a connection string (you'll be showing that as well if you accept my suggestions above). I think you don't need the methods NewMysqlTcpConnetion or NewMySqlUnixSocketConnection (I think those can be deleted), the ones that you want to show are NewMysqlTCPConnectionString and NewMySqlUnixSocketConnectionString (notice that pooling is being set to true in those).

Reply:
Done.